pkg/operator/client/gvkn.go

```diff
@@ -31,7 +31,8 @@ func GetGVKWithNameAndNs(content []byte, baseNS string) (string, OverlySimpleGVK
 		namespace = o.Metadata.Namespace
 	}
 
-	return fmt.Sprintf("%s-%s-%s-%s", o.APIVersion, o.Kind, o.Metadata.Name, namespace), o
+	// kubernetes does not consider the api version when identifying manifests
+	return fmt.Sprintf("%s-%s-%s", o.Kind, o.Metadata.Name, namespace), o
```

Comment:
It's unlikely but possible that two specs with different `APIVersion` will trample each other in the map where this is used a key.  We may need a distinction between what we use in our internal map key and what we use to identify objects that need to be deleted.

Reply:
@sgalsaleh this fix is bit risky in mind. we need to discuss with @marccampbell for a solution.